contrib/generate_precompile.jl

```diff
@@ -19,6 +19,12 @@ CTRL_C = '\x03'
 UP_ARROW = "\e[A"
 DOWN_ARROW = "\e[B"
 
+task = current_task()
```

Comment:
why? IMO, this needs a comment to explain.

Reply:
Because rng state is not initialized at this point and we need it for sorting. I decided to remove what I think is an unnecessary sorting operation later in the file rather than keep the rng initialization.

Reply:
#46278 reintroduced another use of sorting and therefore Random during bootstrapping (via `names`). The number of these problems that have come up now (3) makes me want to fix the issue at the source: we fail to initialize RNG state when the Random stdlib is first loaded. I've patched that in 5b8242144522 with a comment that should hopefully clarify the purpose of those lines.

It would be possible to do this in fewer lines by putting a call to `__init__()` in `stdlib/Random/src/RNG.jl`, but as this code is strictly for bootstrapping, I think it is better to keep in `base/sysimg.jl`.